cmd/server/upsClient.go

```diff
@@ -19,6 +19,59 @@ type upsClient struct {
 
 const BaseUrl = "http://localhost:8080/rest/applications"
 
+func (client *upsClient) deleteIOSVariant(variantId string) bool {
+	variant := client.hasIOSVariant(variantId)
+	if variant != nil {
+		log.Printf("Deleting variant with id `%s`", variant.VariantID)
+
+		url := fmt.Sprintf("%s/%s/adm/%s", BaseUrl, client.config.ApplicationId, variant.VariantID)
+		log.Printf("UPS request", url)
+
+		req, err := http.NewRequest(http.MethodDelete, url, nil)
+
+		httpClient := http.Client{}
+		resp, err := httpClient.Do(req)
+		if err != nil {
+			log.Fatal(err.Error())
+			return false
+		}
+
+		log.Printf("Variant `%s` has been deleted", variant.VariantID)
+		return resp.StatusCode == 204
+	}
+
+	log.Printf("No variant found to delete (Variant Id: `%s`)", variantId)
+	return false
+}
+
+// Find an iOS Variant by it's variant id
+func (client *upsClient) hasIOSVariant(variantId string) *iOSVariant {
+	url := fmt.Sprintf("%s/%s/ios", BaseUrl, client.config.ApplicationId)
```

Comment:
Is the URL the only difference between iOS and Android here? Maybe we can just change it to `hasVariant(type Platform, variantId string) *Variant`? The caller would then cast to pointer to either android or ios.

Reply:
@pb82  I agree with this approach. I was hoping to catch all of this in the refactoring/unifying Jira next. Will I update it now or get it with next body of work?

Reply:
Just to note that there is a lot of duplication while they are being handled differently (googleid vs referenceId) - this will be addressed as I update the apb for android to have the referenceId